### PR TITLE
gpu-viewer: init at 2.26

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

This is a try at packaging [GPU-Viewer](https://github.com/arunsivaramanneo/GPU-Viewer).
~~Currently, the package builds, but when running it I get~~
```
Traceback (most recent call last):
  File "/nix/store/0mzyp5sdg8avd2m2n5ss254px4l21mpq-gpu-viewer-2.26/share/gpu-viewer/Files/GPUViewer.py", line 3, in <module>
    import gi
ModuleNotFoundError: No module named 'gi'
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lovesegfault** [comment]
I don't think you want this `PYTHONPATH` manipulation, just adding the python deps to `propagatedBuildInputs` doesn't work?

**Comment by lovesegfault** [comment]
Works for me with the following diff applied:

```diff
diff --git a/pkgs/applications/misc/gpu-viewer/default.nix b/pkgs/applications/misc/gpu-viewer/default.nix
index 4f489219454..b40db6e6faf 100644
--- a/pkgs/applications/misc/gpu-viewer/default.nix
+++ b/pkgs/applications/misc/gpu-viewer/default.nix
@@ -11,6 +11,10 @@
 , python3
 , wrapGAppsHook
 , gdk-pixbuf
+, lsb-release
+, glxinfo
+, vdpauinfo
+, clinfo
 }:
 
 python3.pkgs.buildPythonApplication rec {
@@ -42,15 +46,16 @@ python3.pkgs.buildPythonApplication rec {
     gdk-pixbuf
   ];
 
-  pythonPath = with python3Packages; [
+  propagatedBuildInputs = with python3Packages; [
     pygobject3
     click
   ];
 
-  postFixup = ''
+  postInstall = ''
     rm $out/bin/gpu-viewer
     makeWrapper ${python3.interpreter} $out/bin/gpu-viewer \
-      --prefix PATH : "${lib.makeBinPath [ vulkan-tools ]}" \
+      --chdir "$out/share/gpu-viewer/Files" \
+      --prefix PATH : "${lib.makeBinPath [ clinfo glxinfo lsb-release vdpauinfo vulkan-tools ]}" \
       --add-flags "$out/share/gpu-viewer/Files/GPUViewer.py" \
       --prefix PYTHONPATH : "$PYTHONPATH" \
       ''${makeWrapperArgs[@]} \
```

**Comment by lovesegfault** [comment]
image

**Comment by GaetanLepage** [comment]
> Works for me with the following diff applied:

Thank you @lovesegfault.
The version I just pushed is slightly different but seems to work now.

**Comment by GaetanLepage** [comment]
> Works for me with the following diff applied:

Thank you @lovesegfault.
The version I just pushed is slightly different but seems to work now.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gpu-viewer</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
Works for me with an Nvidia GTX 950

File: pkgs/applications/misc/gpu-viewer/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  # Prevent double wrapping
+  dontWrapGApps = true;
+
+  postFixup = ''
+    rm $out/bin/gpu-viewer
+    makeWrapper ${python3.interpreter} $out/bin/gpu-viewer \
+      --chdir "$out/share/gpu-viewer/Files" \
+      --prefix PATH : "${lib.makeBinPath [ clinfo glxinfo lsb-release vdpauinfo vulkan-tools ]}" \
+      --add-flags "$out/share/gpu-viewer/Files/GPUViewer.py" \
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gpu-viewer/default.nix:61]:
```suggestion
```

**Comment by GaetanLepage** [comment]
@SuperSandro2000 I tried combining all of your suggestions:
```nix
  postFixup = ''
    rm $out/bin/gpu-viewer
    chmod +x "$out/share/gpu-viewer/Files/GPUViewer.py"
    makeWrapper "$out/share/gpu-viewer/Files/GPUViewer.py" $out/bin/gpu-viewer \
      --prefix PATH : "${lib.makeBinPath [ clinfo glxinfo lsb-release vdpauinfo vulkan-tools ]}" \
      --chdir "$out/share/gpu-viewer/Files" \
      ''${makeWrapperArgs[@]} \
      ''${gappsWrapperArgs[@]}
  '';
```

However, it doesn't work: the process exits with 255.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**